### ISSUE TITLE
open firmware

### ISSUE THREAD

**Issue by raszpl** [labs-lv/blasterboard#1]
https://www.vogons.org/viewtopic.php?p=701465#p701465

2018-09-24, 16:33 LABS:
>I will make an open firmware

Any news on that front?
btw last year in the Hackaday comment you wrote
>I’m in the process of translating the firmware to a human-readable form, about 80% is ready.

were you talking about the clone firmware from snark-barker repository, or did you manage to get your hands on the genuine  Creative 2.01 firmware? schlae never shared his dump :(

The only other firmware available I know of is "ATI Stereo F/X ISA Sound Card" dumped by MAME people
https://forums.bannister.org/ubbthreads.php?ubb=showflat&Number=82634&page=4
https://archive.org/download/MESS-0.151.BIOS.ROMs/MESS-0.151.BIOS.ROMs.zip/MESS%200.151%20ROMs%2Fstereo_fx.zip
more about this card here https://sites.google.com/site/soundcardcomparison/

Someone also claimed to have dumped CT1600 Sound Blaster pro 2  MHS CT1341 chip, but never posted it anywhere
https://forums.bannister.org/ubbthreads.php?ubb=showflat&Number=95856#Post95856

**Comment by uggima** [comment]
Yeah, got to say I would LOVE to see that atmega328 source (as long as its not just assembly :D). I recently got a 486 laptop to re-play some retro games etc.... then realised soundcards weren't exactly standard equipment for laptops of the era.....
But, it does have a pcmcia card slot.... soooo, why not make a sound card for it?
Bandwidth shouldn't be quite as limited as LPT port and can't think of any other possible issues. 
Well, if you do feel like releasing your code that would be a REAL help, sure I could write "something"... either way its going to be a while before anything gets made.

**Comment by raszpl** [comment]
> see that atmega328 source (as long as its not just assembly :D)

? https://github.com/labs-lv/blasterboard/tree/master/Firmware/2.1-current/src

My post is in regards to original SB 2.01

**Comment by labs-lv** [comment]
Hello, guys

razspl, sorry for very late answer - missed your first post. 
I doubt Snark Barker uses any kind of a clone firmware. It actually uses the original SB 2.01 firmware. Why would you think it uses the clone firmware? After disassembly it is obvious that all the hidden, comapared to the official documentation, functionality that opens up could not be written by anyone except the SB developers. Just too many service code in there. And yes, I disassembled that exact firmware of a Snark Barker. I have it with some comments added by me just for easier analysis of functions under interest during the development of BlasterBoard. If you want-I can e-mail it to you as releasing it to the public will not be convenient in my opinion.

uggima, BlasterBoard's firmware is already open-source and is a pure AVR assembler, the link is given by razspl above.

**Comment by raszpl** [comment]
snark-barker firmware file

```
xxd -r -p sb.hex > sb.bin
strings sb.bin

lot of garbage
(C)19k
92 Anchor Electronics Co.,
```

Couple of days after this thread tubetime uploaded real SB firmware 2.02 https://github.com/schlae/sb-firmware so its more or less solved, unless you are in a possession of SB 2.01 dsp firmware?
